flang/lib/Lower/OpenMP.cpp

```diff
@@ -178,14 +179,39 @@ genOMP(Fortran::lower::AbstractConverter &converter,
                          &clause.u)) {
         const Fortran::parser::OmpObjectList &ompObjectList = copyinClause->v;
         genObjectList(ompObjectList, converter, copyinClauseOperands);
+      } else if (const auto &allocateClause =
+                     std::get_if<Fortran::parser::OmpAllocateClause>(
+                         &clause.u)) {
+        mlir::Value allocator;
+        const Fortran::parser::OmpObjectList &ompObjectList =
+            std::get<Fortran::parser::OmpObjectList>(allocateClause->t);
+        int count = ompObjectList.v.size();
+        const auto &allocatorValue = std::get<
+            std::optional<Fortran::parser::OmpAllocateClause::Allocator>>(
+            allocateClause->t);
+        if (allocatorValue) {
+          allocator = fir::getBase(converter.genExprValue(
+              *Fortran::semantics::GetExpr(allocatorValue->v)));
+          while (count--) {
```

Comment:
Will any of the usage in the following link work here for small vector?
https://stackoverflow.com/questions/30998444/add-same-value-multiple-times-to-stdvector-repeat

Reply:
Thank you for the link Kiran, I'll make sure to give this a go